File: Codeforces/2199A.kt
fun main() {
    val t = readLine()!!.toInt()

    repeat(t) {
        val k = readLine()!!.toInt()

        val (a1, b1) = readLine()!!.split(" ").map { it.toInt() }
        val (a2, b2) = readLine()!!.split(" ").map { it.toInt() }

        val b3Max = minOf(k, 3 * k - (b1 + b2))

        val bTotal = b1 + b2 + b3Max
        val aTotal = a1 + a2

        if (bTotal > aTotal) {
            println("yes")
        } else {
            println("no")
        }
    }
}

File: Codeforces/2199A.py
def main():
    for _ in range(int(input())):
        k = int(input())
        a1, b1 = map(int, input().split())
        a2, b2 = map(int, input().split())

        b3_max = min(k, 3 * k - (b1 + b2))

        b_total = b1 + b2 + b3_max
        a_total = a1 + a2

        if b_total > a_total:
            print('yes')
        else:
            print('no') 


if __name__ == "__main__":
    main()
